src/odemis/util/angleres.py

```diff
@@ -874,9 +874,32 @@ def project_angular_spectrum_to_grid(
         omega_list_short = omega_list_raw[first_px: last_px + 1]
         d = data[first_px: last_px + 1, wli]
         d_corrected = d / omega_list_short
-        # Pad with 0.0 for points outside (left and right) the source data's angular range.
+        # Pad with numpy.nan for points outside (left and right) the source data's angular range.
         # This prevents creating artificial signal at the edges from clamping.
-        data_lin[:, wli] = numpy.interp(theta_list_linear, theta_list_short, d_corrected, left=0.0, right=0.0)
+        data_lin[:, wli] = numpy.interp(theta_list_linear, theta_list_short, d_corrected, left=numpy.nan, right=numpy.nan)
+
```

Comment:
Please add a comment here explaining the goal of this whole part. Alternatively, you could even put this part into its own function (as it's pretty much independent from the rest of the variables) and put the comment as the docstring of the function.